spec/features/settings/dex_connector_oidc_feature_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
+
+# rubocop:disable RSpec/ExampleLength
+describe "Feature: OIDC connector settings", js: true do
+  let!(:user)                { create(:user) }
+  let!(:dex_connector_oidc)  { create(:dex_connector_oidc, :skip_validation) }
+  let!(:dex_connector_oidc2) { create(:dex_connector_oidc, :skip_validation) }
+  let!(:dex_connector_oidc3) { create(:dex_connector_oidc, :skip_validation) }
+
+  before do
+    setup_done
+    login_as user, scope: :user
+  end
+
+  describe "#index" do
+    before do
+      VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
+        visit settings_dex_connector_oidcs_path
+      end
+    end
+
+    it "allows a user to delete an oidc connector" do
+      expect(page).to have_content(dex_connector_oidc.name)
+      accept_alert do
+        find(".dex_connector_oidc#{dex_connector_oidc.id} .delete-btn").click
+      end
+
+      expect(page).to have_content("OIDC Connector was successfully removed.")
+      expect(page).not_to have_content(dex_connector_oidc.name)
+    end
+
+    it "allows a user to go to an oidc connector's details page" do
+      click_link(dex_connector_oidc.name)
+
+      expect(page).to have_current_path(settings_dex_connector_oidc_path(dex_connector_oidc))
+    end
+
+    it "allows a user to go to an oidc connector's edit page" do
+      find(".dex_connector_oidc#{dex_connector_oidc.id} .edit-btn").click
+
+      expect(page).to have_current_path(edit_settings_dex_connector_oidc_path(dex_connector_oidc))
+    end
+
+    it "allows a user to go to the new oidc connector page" do
+      click_link("Add OIDC connector")
+
+      expect(page).to have_current_path(new_settings_dex_connector_oidc_path)
+    end
+
+    it "lists all the oidc connectors" do
+      expect(page).to have_content(dex_connector_oidc.name)
+      expect(page).to have_content(dex_connector_oidc2.name)
+      expect(page).to have_content(dex_connector_oidc3.name)
+    end
+  end
+
+  describe "#new" do
+    before do
+      visit new_settings_dex_connector_oidc_path
+    end
+
+    it "allows a user to create an OIDC connector" do
+      VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
```

Comment:
Is a VCR needed? Reviewing the changes I saw that the other VCR's were able to be done away with. Is it possible to do similar here?

Reply:
Sure is.

The others don't need it because the data in those cases is being directly inserted into the database by a known-good, static source, so exercising the connection validation was redundant.  This one actually needs VCR, since it's testing the actual "user clicks validate before save" flow in a simulated browser, which takes data from the form.  This is also known-good data, but there's no way to bypass the validation in this one because it's coming in from the web form instead of direct to the DB via the factory.

Reply:
Ok understood. Thank you for the explanation.